File: src/s100-disvet/s100-disvet/MyGlobal.cs
using System;
using System.Collections.Generic;
using System.Text;
using System.IO;
using System.Data.SQLite;
using System.Security.Cryptography;

public struct tFiles
{
    public string paramName_a;
    public string nombre_a;
    public string tipo_a;
    public byte[] file_a;
}    

namespace s100_disvet
{
    class MyGlobal
    {

        public static string getMD5Hash(string input)
        {
            // Create a new instance of the MD5CryptoServiceProvider object.
            MD5 md5Hasher = MD5.Create();

            // Convert the input string to a byte array and compute the hash.
            byte[] data = md5Hasher.ComputeHash(Encoding.Default.GetBytes(input));

            // Create a new Stringbuilder to collect the bytes
            // and create a string.
            StringBuilder sBuilder = new StringBuilder();

            // Loop through each byte of the hashed data 
            // and format each one as a hexadecimal string.
            for (int i = 0; i < data.Length; i++)
            {
                sBuilder.Append(data[i].ToString("x2"));
            }

            // Return the hexadecimal string.
            return sBuilder.ToString();
        }

        public static long UnixTimeNow()
        {
            DateTime origin = new DateTime(1970, 1, 1, 0, 0, 0, 0);
            TimeSpan diff = DateTime.UtcNow - origin;

            return (long)diff.TotalSeconds;

        }


        public static byte[] ReadByteArrayFromFile(string fileName)
        {
            byte[] buff = null;

            FileStream fs = new FileStream(fileName, FileMode.Open, FileAccess.Read);

            BinaryReader br = new BinaryReader(fs);
            long numBytes = new FileInfo(fileName).Length;
            buff = br.ReadBytes((int)numBytes);

            fs.Close();
            return buff;
        }


    }
}

File: src/s100-tpm/s100-tpm/MyGlobal.vb
Imports System.Data.SQLite
Public Class MyGlobal
    Public Shared sqlConnMaestros_a As SQLiteConnection
    Public Shared sqlConnTransacciones_a As SQLiteConnection

    Public Shared idusuario As String
    Public Shared iddispositivo As String = "pda001"

    Public Shared MyTime As String
    Public Shared MyLatitud As Double
    Public Shared MyLongitud As Double
    Public Shared MyHeading As Double
    Public Shared MySpeed As Double
    Public Shared MyAltitude As Double

    Public Shared Function UnixTimeNow() As Long
        Dim _TimeSpan As TimeSpan = DirectCast((DateTime.UtcNow - New DateTime(1970, 1, 1, 0, 0, 0)), TimeSpan)
        Return CLng(_TimeSpan.TotalSeconds)
    End Function

    Public Shared Function getMD5Hash(ByVal strToHash As String) As String
        Dim md5Obj As New Security.Cryptography.MD5CryptoServiceProvider
        Dim bytesToHash() As Byte = System.Text.Encoding.ASCII.GetBytes(strToHash)

        bytesToHash = md5Obj.ComputeHash(bytesToHash)

        Dim strResult As String = ""

        For Each b As Byte In bytesToHash
            strResult += b.ToString("x2")
        Next

        Return strResult
    End Function
End Class
